src/plugins/console/public/lib/utils/utils.test.js

```diff
@@ -82,4 +82,65 @@ describe('Utils class', () => {
       'e", f"',
     ]);
   });
+
+  describe('formatRequestBodyDoc', function () {
+    const tests = [
+      {
+        source: ['{\n  "test": {}\n}'],
+        indent: false,
+        assert: ['{"test":{}}'],
+      },
+      {
+        source: ['{"test":{}}'],
+        indent: true,
+        assert: ['{\n  "test": {}\n}'],
+      },
+      {
+        source: ['{\n  "test": """a\n  b"""\n}'],
+        indent: false,
+        assert: ['{"test":"a\\n  b"}'],
+      },
+      {
+        source: ['{"test":"a\\n  b"}'],
+        indent: true,
+        assert: ['{\n  "test": """a\n  b"""\n}'],
+      },
+    ];
+
+    tests.forEach(({ source, indent, assert }, id) => {
+      test(`Test ${id}`, () => {
+        const formattedData = utils.formatRequestBodyDoc(source, indent);
+        expect(formattedData.data).toEqual(assert);
+      });
+    });
+  });
+
+  describe('hasComments', function () {
+    const tests = [
```

Comment:
Thanks for adding these tests! It's a bit hard for me to understand the intention behind each of them, though. Can we group them into `describe` blocks with descriptive names, and also add more meaningful info into the `test` description? Here's how I suggest you refactor them, with a couple new sets of tests that I also suggest you add:

```js
    const runCommentTests = tests => {
      tests.forEach(({ source, assert }, id) => {
        test(`\n${source}`, () => {
          const hasComments = utils.hasComments(source);
          expect(hasComments).toEqual(assert);
        });
      });
    };

    describe('ignore comment tokens as values', () => {
      runCommentTests([
        { source: '{\n  "test": {\n    "f": "//"\n  }\n}', assert: false },
        { source: '{\n  "test": {\n    "f": "/* */"\n  }\n}', assert: false },
        { source: '{\n  "test": {\n    "f": "#"\n  }\n}', assert: false },
      ]);
    });

    describe('ignore comment tokens as field names', () => {
      runCommentTests([
        { source: '{\n  "#": {\n    "f": {}\n  }\n}', assert: false },
        { source: '{\n  "//": {\n    "f": {}\n  }\n}', assert: false },
        { source: '{\n  "/* */": {\n    "f": {}\n  }\n}', assert: false },
      ]);
    });
```

I think this will help a) describe the behavior under test and b) emit more meaningful failure messages, like this:

```
  ● Utils class › hasComments › ignore comment tokens as field names › 
{
  "/* */": {
    "f": {}
  }
}
```

Regarding the failures of the tests I've suggested, I believe this indicates a bug in the auto-indent behavior of this feature. You can reproduce by trying to auto-indent with the below request. Instead of collapsing and expanding, nothing happens.

```
POST test/_doc
{
  "b": {
    "#": {
      "a": "/*"
    }
  }
}

# This is the expected collapsed result.
POST test/_doc
{"b":{"#":{"a":"/*"}}}
```

Because this is a non-destructive bug in an edge case in an entirely new feature, I'd be happy to unblock this PR by temporarily disabling these new tests and tracking the bug in an issue.

Reply:
Thank you for the suggestions! I've skipped these tests above for now. It seems like the regex pattern we are using is not ignoring the comments in quotes. Before merging this PR, I'll create a follow-up issue to tackle this bug.

Reply:
Created a separate issue for this https://github.com/elastic/kibana/issues/135186